library/Icinga/Data/UserSuggestions.php

```diff
@@ -0,0 +1,154 @@
+use Icinga\User;
+use Icinga\Web\Notification;
+use Icinga\Data\Filter\Filter;
+use ipl\Web\Control\SimpleSuggestions;
+
+class UserSuggestions extends SimpleSuggestions
+{
+    /** @var mixed */
+    protected $backends;
+
+    /** @var string The name of user group */
+    protected $userGroupName;
+
+    /** @var mixed The user group backend */
+    protected $userGroupBackend;
+
+    /**
+     * Set User group backend
+     *
+     * @param mixed $userGroupBackend
+     */
+    public function setUserGroupBackend($userGroupBackend): self
+    {
+        $this->userGroupBackend = $userGroupBackend;
+
+        return $this;
+    }
+
+    /**
+     * Get User group backend
+     *
+     * @return mixed $userGroupBackend
+     */
+    public function getUserGroupBackend()
+    {
+        return $this->userGroupBackend;
+    }
+
+    /**
+     * Set User group name
+     *
+     * @param string $userGroupName
+     */
+    public function setUserGroupName(string $userGroupName): self
+    {
+        $this->userGroupName = $userGroupName;
+
+        return $this;
+    }
+
+    /**
+     * Get User group name
+     *
+     * @return string $userGroupName
+     */
+    public function getUserGroupName(): string
+    {
+        return $this->userGroupName;
+    }
+
+    /**
+     * Get backends
+     *
+     * @return mixed
+     */
+    public function getBackends()
+    {
+        return $this->backends;
+    }
+
+    /**
+     * Set backends
+     *
+     * @param mixed $backends
+     */
+    public function setBackends($backends): self
+    {
+        $this->backends = $backends;
+
+        return $this;
+    }
+
+    protected function fetchSuggestions(string $searchTerm, array $exclude)
+    {
+        $count = 0;
+        foreach ($this->getBackends() as $backend) {
+            try {
+                if ($backend instanceof DomainAwareInterface) {
+                    $domain = $backend->getDomain();
+                } else {
+                    $domain = null;
+                }
+
+                $members = $this->getUserGroupBackend()
+                    ->select()
+                    ->from('group_membership', ['user_name'])
+                    ->where('group_name', $this->getUserGroupName())
+                    ->fetchColumn();
+
+
+                if (! empty($exclude)) {
+                    $members = array_merge($members, $exclude);
+                }
+
+                $filter = Filter::matchAll(
+                    Filter::where('user_name', $searchTerm),
+                    Filter::not(Filter::where('user_name', $members))
+                );
+
+                $users = $backend->select(['user_name'])
+                    ->limit(self::DEFAULT_LIMIT)
+                    ->applyFilter($filter)
+                    ->fetchColumn();
+
+                foreach ($users as $userName) {
+                    if ($count === self::DEFAULT_LIMIT) {
+                        return;
+                    }
```

Comment:
Why is this necessary? The `SimpleSuggestions` class should take care of that as does the `Suggestions` class for the search bar.